clang/utils/TableGen/ClangAttrEmitter.cpp

```diff
@@ -3664,10 +3664,12 @@ static void GenerateMutualExclusionsChecks(const Record &Attr,
   if (Attr.isSubClassOf("TypeAttr"))
     return;
 
-  // This means the attribute is either a statement attribute or a decl
-  // attribute, find out which.
+  // This means the attribute is either a statement attribute, a decl
+  // attribute, or both; find out which.
   bool CurAttrIsStmtAttr =
       Attr.isSubClassOf("StmtAttr") || Attr.isSubClassOf("DeclOrStmtAttr");
+  bool CurAttrIsDeclAttr =
+      !CurAttrIsStmtAttr || Attr.isSubClassOf("DeclOrStmtAttr");
```

Comment:
@AaronBallman, thanks for the cleanup logic. Could you please explain why we can not use same for other statements attrs?

I am not much familiar with tablegen.  I apologize if i understand wrong. 

 Example: def : MutualExclusions<[SSYCLIntelFPGAMaxInterleavingAttr,
                        SYCLIntelFPGADisableLoopPipelining]>;

Reply:
There two things with this.

This particular change is recognizing that statement and declaration attributes are not mutually exclusive -- an attribute can be *both* a decl and a stmt attr. So it changes the `else` into an `if`.

However, this won't be sufficient to make this feature work for us quite yet because it turns out that attributed statements work differently than declarations and so my speculative work for mutually exclusive statement attributes was... wrong. I'm working on a fix for that, but it'll likely not happen today.

That's why statement attributes are not being handled as part of this patch.

Reply:
Thanks @AaronBallman for the explanation.